src/main/scala/io/iohk/ethereum/vm/EvmConfig.scala

```diff
@@ -0,0 +1,177 @@
+    If Some(x): let limit = GAS * (x - 1) / x; let CALL's gas = min(requested, limit). let CREATE's gas = limit.
+    If None: let CALL's gas = (requested > GAS ? [OOG] : GAS). let CREATE's gas = GAS
+   */
+  val PostEIP150Config = HomesteadConfig.copy(
+    feeSchedule = FeeSchedule.PostEIP150FeeSchedule,
+    subGasCapDivisor = Some(64))
+
+  val PostEIP160Config = PostEIP150Config.copy(
+    feeSchedule = FeeSchedule.PostEIP160FeeSchedule)
+
+  private val transitionBlockToConfigMapping: Map[BigInt, EvmConfig] = Map(
+    Config.Blockchain.frontierBlockNumber -> FrontierConfig,
+    Config.Blockchain.homesteadBlockNumber -> HomesteadConfig,
+    Config.Blockchain.eip150BlockNumber -> PostEIP150Config,
+    Config.Blockchain.eip160BlockNumber -> PostEIP160Config)
+
+}
+
+case class EvmConfig(
+    feeSchedule: FeeSchedule,
+    opCodes: List[OpCode],
+    exceptionalFailedCodeDeposit: Boolean,
+    maxMemory: UInt256 = UInt256(Long.MaxValue), /* used to artificially limit memory usage by incurring maximum gas cost */
+    maxCallDepth: Int = 1024,
+    subGasCapDivisor: Option[Long] = None) {
+
+  val byteToOpCode: Map[Byte, OpCode] =
+    opCodes.map(op => op.code -> op).toMap
+}
+
+object FeeSchedule {
+
+  // See YP, appendix G
+  sealed trait GasCost
+  object GasCost {
+    case object G_zero extends GasCost
+    case object G_base extends GasCost
+    case object G_verylow extends GasCost
+    case object G_low extends GasCost
+    case object G_mid extends GasCost
+    case object G_high extends GasCost
+    case object G_extcode extends GasCost
+    case object G_balance extends GasCost
+    case object G_sload extends GasCost
+    case object G_jumpdest extends GasCost
+    case object G_sset extends GasCost
+    case object G_sreset extends GasCost
+    case object R_sclear extends GasCost
+    case object R_selfdestruct extends GasCost
+    case object G_selfdestruct extends GasCost
+    case object G_create extends GasCost
+    case object G_codedeposit extends GasCost
+    case object G_call extends GasCost
+    case object G_callvalue extends GasCost
+    case object G_callstipend extends GasCost
+    case object G_newaccount extends GasCost
+    case object G_exp extends GasCost
+    case object G_expbyte extends GasCost
+    case object G_memory extends GasCost
+    case object G_txcreate extends GasCost
+    case object G_txdatazero extends GasCost
+    case object G_txdatanonzero extends GasCost
+    case object G_transaction extends GasCost
+    case object G_log extends GasCost
+    case object G_logdata extends GasCost
+    case object G_logtopic extends GasCost
+    case object G_sha3 extends GasCost
+    case object G_sha3word extends GasCost
+    case object G_copy extends GasCost
+    case object G_blockhash extends GasCost
+    case object G_extcodesize extends GasCost
+    case object G_extcodecopy_base extends GasCost
+    case object G_selfdestruct_to_new_account extends GasCost
```

Comment:
I see why you did that. The thing is all those "keys" correspond directly the fee schedule defined in the YP appending G. And the definition of SELFDESTRUCT refers to `G_newaccount` (which is the same as for CALL, and that's in line with EIP150). 

It really makes it easier to reason about code vs specification when the names correspond, so if you could find a way for SELFDESTRUCT to utilise `G_newaccount` (a boolean flag?) that would be preferred, IMO.